File: backend/src/routes/announcement.js
import express from "express";
import { requireAuth, requireRole } from "../middleware/auth.js";
import { ROLES } from "../constants/roles.js";
import { generateAnnouncement } from "../services/announcementService.js";

const router = express.Router();

/**
 * GET /api/announcement?date=YYYY-MM-DD
 * Returns a markdown-formatted announcement string for the given date.
 */
router.get(
    "/",
    requireAuth,
    requireRole([ROLES.ADMIN, ROLES.LOGISTICS]),
    (req, res) => {
        const { date } = req.query;
        if (!date) {
            return res.status(400).json({ error: "date query param is required" });
        }

        const text = generateAnnouncement(date);
        res.json({ date, text });
    }
);

export default router;

File: mhp-python/app/routes/announcement.py
from fastapi import APIRouter, Depends
from datetime import date as DateType

from app.auth import require_roles
from app.constants import Roles
from app.services.announcement_service import generate_announcement

router = APIRouter()


@router.get("")
def announcement(date: str = None, user: dict = Depends(require_roles([Roles.ADMIN, Roles.LOGISTICS]))):
    target = date or DateType.today().isoformat()
    text = generate_announcement(target)
    return {"date": target, "text": text}
